documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -2234,3 +2238,83 @@ Eg:
 ```
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4005" ></a>R4005 XmsEnumNameUnique
+
+**Category** : ARM Warning
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : Must not have duplicate name in x-ms-enum extension , make sure every x-ms-enum name unique.  
+
+**Description** : Property name in x-ms-extension is required , the rule introduce to avoid duplicate x-ms-enum name.
+
+**CreatedAt**: March 18, 2020
+
+**LastModifiedAt**: March 18, 2020
+
+**How to fix the violation**: Update the duplicate x-ms-enum name :
+
+The following would be invalid:
+```json
+    "State": {
+        "description": "The  state of the configuration store.",
+        "enum": [
+            "Failed",
+            "Canceled"
+        ],
+        "type": "string",
+        "readOnly": true,
+        "x-ms-enum": {
+            "name": "DuplicateName",
+            "modelAsString": true
+        }
+    },
+    "status": {
+        "description": "The state code.",
+        "enum": [
+            "Success",
+            "FAILED"
+        ],
+        "readOnly": true,
+        "type": "string",
+        "x-ms-enum": {
+            "name": "DuplicateName",
+            "modelAsString": true
+        }
+    }
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4006" ></a>R4006 XmsCodeGenerationSettingDeprecated
+
+**Category** : ARM Warning
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : The x-ms-code-generation-settings extenison was deprecated. Consider remove it from the swagger.  
+
+**Description** : New autorest (V3) will stop supporting “x-ms-code-generation-settings” inside the Swagger. This was designed before there was Readme.md to hold configuration options, and this node should disappear and be replaced by actual readme descriptions.
```

Comment:
My proposal:

"x-ms-code-generation-settings” is being deprecated. AutoRest (v3) is using settings in readme file for code generation. Please ensure to remove the parameter from swagger spec and move settings to readme.

Reply:
how about: 
x-ms-code-generation-settings is being deprecated. AutoRest (v3) is using settings in readme file for code generation and will stop supporting it inside the swagger file. Please ensure to remove the parameter from swagger spec and move settings to readme.

I think it's necessary to point that autorest will stop spport it